src/escalus_bosh.erl

```diff
@@ -240,27 +241,33 @@ init([Args, Owner]) ->
     Port = proplists:get_value(port, Args, 5280),
     Path = proplists:get_value(path, Args, <<"/http-bind">>),
     Wait = proplists:get_value(bosh_wait, Args, ?DEFAULT_WAIT),
+    Timeout = proplists:get_value(timeout, Args, infinity),
```

Comment:
`infinity` is never a good timeout value. We should approximate infinity with a long, but still finite, number here.